changelog/unreleased/issue-4983

```diff
@@ -0,0 +1,8 @@
+Enhancement: add SLSA provenance to the Docker images
+
+Restic's Docker image build workflow now includes SLSA provenance generation. 
```

Comment:
This only applies to the images on GHCR, not the images on Dockerhub.

Reply:
I am not sure how the build-and-push steps work to upload to Dockerhub. Let me see if I can make this a generic solution which would work for any repository, not limited to Dockerhub and GHCR.